src/shm_manager.h

```diff
@@ -179,6 +179,7 @@ class SharedMemoryManager {
                                        shm_ownership_data](T* memory) {
       bool destroy = false;
       bi::scoped_lock<bi::interprocess_mutex> gaurd{*shm_mutex_};
+      GrowIfNeeded(0);
```

Comment:
Can you add comment on this? By the name it seems like you are relying on the side-affect of the function?

Reply:
Added a comment. Yes, basically, both sides (stub and main process) need to make sure to use the updated shared memory map for allocation/deallocation.

Reply:
Actually, why not using the safe version of `Deallocate`? To avoid mutex dead lock, you can add scope:
```
      ...
      {
        bi::scoped_lock<bi::interprocess_mutex> gaurd{*shm_mutex_};
        shm_ownership_data->ref_count_ -= 1;
        if (shm_ownership_data->ref_count_ == 0) {
          destroy = true;
        }
      }
      if (destroy) {
        Deallocate(handle);
      }
      ...
```

Reply:
It is to avoid locking the mutex two times.